File: src/main/java/com/api/util/FileSystem.java
package com.api.util;

import java.io.File;
import java.io.IOException;
import java.net.URL;
import java.util.UUID;
import org.apache.commons.io.FileUtils;


/**
 * Provides working with all used application files.
 */
public class FileSystem
{
	private static final String tempCaptchaPath = "/cache/temp_captcha/";



	public static File downloadCaptchaToFile(String captchaURL) throws IOException
	{
		File captchaImageFile = new File (tempCaptchaPath+"/"+UUID.nameUUIDFromBytes(captchaURL.getBytes()).toString());
		
	//	captchaImageFile.getParentFile().mkdirs();
	//	captchaImageFile.createNewFile();
		
		FileUtils.copyURLToFile(new URL(captchaURL), captchaImageFile);
		
		return captchaImageFile;
	}
	

}

File: src/main/java/com/api/util/FileSystem.kt
package com.api.util

import org.apache.commons.io.FileUtils
import java.io.File
import java.net.URL
import java.util.*

object FileSystem {
    private const val tempCaptchaPath = "/cache/temp_captcha/"

    fun downloadCaptchaToFile(captchaURL: String): File {
        val captchaImageFile = File(tempCaptchaPath + "/" + UUID.nameUUIDFromBytes(captchaURL.toByteArray()).toString())
        FileUtils.copyURLToFile(URL(captchaURL), captchaImageFile)
        return captchaImageFile
    }
}
